ripple/src/main/webapp/businesscarousels.js

```diff
@@ -80,99 +80,36 @@ var maxWalkingTime = 1200;
 
 /* Code to dynamically load carousels. Calls addDynamicCarousel function to populate the carousels. */
 function loadCarousels() {
-    addDynamicCarouselByTag("black-owned-businesses", "Black-owned");
-    addDynamicCarouselByTime("under-20-mins-away", maxWalkingTime);
-    addDynamicCarouselByTag("trending-near-you", "Trending");
-    addDynamicCarouselByTag("up-and-coming", "New");
+    addDynamicCarousel("black-owned-businesses", "Black-owned", false);
+    addDynamicCarousel("under-20-mins-away", false, maxWalkingTime);
+    addDynamicCarousel("trending-near-you", "Trending", false);
+    addDynamicCarousel("up-and-coming", "New", false);
 }
 
 /* Dynamically loads content into Bootstrap carousel by reconstructing HTML elements and making a Firestore query. 
-   Filters by tag.*/
-function addDynamicCarouselByTag(carouselId, tag) {
+   Filters by tag or walking time in seconds.*/
+function addDynamicCarousel(carouselId, tag, time) {
     db.collection("businesses").get().then((querySnapshot) => {
         var makeElement = false;
         var firstCard = true;
         querySnapshot.forEach((doc) => {
-            // Check if the city in Firestore matches the city extracted from the user inputted address.
-            // Also checks if the business's tag list contains the tag that the carousel requires.
-            if (doc.data().address != null && doc.data().address[1] == localStorage.getItem('enteredCity') 
-                && doc.data().tags[convertToRawString(tag)] == tag) {
-              makeElement = true;
-              const card = document.createElement('div');
-              card.classList = 'carousel-inner row w-100 mx-auto';
-              
-              // Distance matrix calculations
-              // Get user's latitude and longitude from localStorage
-              var userLat = localStorage.getItem('enteredLat');
-              var userLong = localStorage.getItem('enteredLong');
-
-              // Get business latitude and longitude from GeoPoint object in firestore
-              var busLat = doc.data().coordinates.latitude;
-              var busLong = doc.data().coordinates.longitude;
-
-              // Reconstruct as maps LatLng object
-              var origin = new google.maps.LatLng(userLat, userLong);
-              var destination = new google.maps.LatLng(busLat, busLong);
-              
-              // Call distance matrix service
-              var service = new google.maps.DistanceMatrixService();
-              service.getDistanceMatrix({
-                origins: [origin],
-                destinations: [destination],
-                travelMode: google.maps.TravelMode.WALKING,
-              }, callback);
```

Comment:
Is the maps function calculating walking time/distance generalized somewhere else since it's being deleted here?

Reply:
I deleted the whole function because I combined both tags and time filtering into one.